src/coreclr/src/jit/lsra.cpp

```diff
@@ -4435,7 +4439,26 @@ void LinearScan::spillGCRefs(RefPosition* killRefPosition)
         {
             continue;
         }
-        unassignPhysReg(regRecord, assignedInterval->recentRefPosition);
+        bool needsKill = varTypeIsGC(assignedInterval->registerType);
```

Comment:
Looks like I am missing something, the prev line has:
```
if (!varTypeIsGC(assignedInterval->registerType)))
  continue;
```
`varTypeIsGC(assignedInterval->registerType))` is always true after that, means:

`bool needsKill = varTypeIsGC(assignedInterval->registerType); ` always true.
`if (!needsKill)` always false, so the code and the comment under that block is never executed.

Reply:
That was a mistake in extracting the fix from my EHWriteThru branch to this one. Thanks for catching that!